cpp/oneapi/dal/algo/jaccard/backend/cpu/vertex_similarity_default_kernel_hsw.cpp

```diff
@@ -41,8 +271,78 @@ vertex_similarity_result call_jaccard_default_kernel<undirected_adjacency_array_
+        int *first_vertices = reinterpret_cast<int *>(result_ptr);
+        int *second_vertices = first_vertices + number_elements_in_block;
+        float *jaccard = reinterpret_cast<float *>(first_vertices + 2 * number_elements_in_block);
+        std::int64_t nnz = 0;
+        for (std::int32_t i = row_begin; i < row_end; ++i) {
+            const auto i_neighbor_size = g_degrees[i];
+            const auto i_neigbhors = g_vertex_neighbors + g_edge_offsets[i];
+            const auto diagonal = min(i, column_end);
+            for (std::int32_t j = column_begin; j < diagonal; j++) {
+                const auto j_neighbor_size = g_degrees[j];
+                const auto j_neigbhors = g_vertex_neighbors + g_edge_offsets[j];
+                if (!(i_neigbhors[0] > j_neigbhors[j_neighbor_size - 1]) &&
+                    !(j_neigbhors[0] > i_neigbhors[i_neighbor_size - 1])) {
+                    auto intersection_value =
+                        intersection(i_neigbhors, j_neigbhors, i_neighbor_size, j_neighbor_size);
+                    if (intersection_value) {
+                        jaccard[nnz] =
+                            float(intersection_value) /
+                            float(i_neighbor_size + j_neighbor_size - intersection_value);
+                        first_vertices[nnz] = i;
+                        second_vertices[nnz] = j;
+                        nnz++;
+                    }
+                }
+            }
+
+            if (diagonal >= column_begin && diagonal < column_end) {
+                jaccard[nnz] = 1.0;
+                first_vertices[nnz] = i;
+                second_vertices[nnz] = diagonal;
+                nnz++;
+            }
+
+            for (std::int32_t j = max(column_begin, diagonal + 1); j < column_end; j++) {
+                const auto j_neighbor_size = g_degrees[j];
+                const auto j_neigbhors = g_vertex_neighbors + g_edge_offsets[j];
+                if (!(i_neigbhors[0] > j_neigbhors[j_neighbor_size - 1]) &&
+                    !(j_neigbhors[0] > i_neigbhors[i_neighbor_size - 1])) {
+                    auto intersection_value =
+                        intersection(i_neigbhors, j_neigbhors, i_neighbor_size, j_neighbor_size);
+                    if (intersection_value) {
+                        jaccard[nnz] =
+                            float(intersection_value) /
+                            float(i_neighbor_size + j_neighbor_size - intersection_value);
+                        first_vertices[nnz] = i;
+                        second_vertices[nnz] = j;
+                        nnz++;
+                    }
+                }
+            }
+        }
+        vertex_similarity_result res(
+            homogen_table::wrap(first_vertices, 2, number_elements_in_block),
+            homogen_table::wrap(jaccard, 1, number_elements_in_block),
+            nnz);
+        return res;
+    }
+} // namespace detail
 } // namespace detail
```

Comment:
I'm pretty sure that in this exact version there were two `} // namespace detail` lines.

Reply:
But it was fixed later

Reply:
Ok, thanks, it was fixed in the next commits